### PR TITLE
Update to RSpec 3.3 in suspenders apps

### PR DESCRIPTION
**Pull request by derekprior**
RSpec 3.3 added some very handy commands for filtering examples. With
the addition of the status persistence file configuration we can now:

* run `rspec --only-failures` to re-run only the examples that failed.
* run `rspec --next-failure` to re-run only the first failing test from
  your previous run. If that passes it will run the successive failures
  until one of them fails again

Additionally, rspec 3.3 includes stable random ordering (even when you
add new examples), the ability to bisect failures when debugging
ordering-dependent failures, better failure output...

http://rspec.info/blog/2015/06/rspec-3-3-has-been-released/

### CODE REVIEW

**Comment by teoljungberg** [comment]
I'm into this

**Comment by calleerlandsson** [comment]
Cool! I'm in favor.

File: templates/spec_helper.rb [hunk 1]
```diff
@@ -16,6 +16,7 @@
     mocks.verify_partial_doubles = true
   end
 
+  config.example_status_persistence_file_path = "tmp/examples.txt"
```
Comment (tute) [inline templates/spec_helper.rb]: Let's call this file `tmp/rspec_examples.txt`?

**Comment by tute** [comment]
:+1:

**Comment by mcmire** [comment]
Oh man! I'm glad they're adding support for re-running failed tests. This is an idea I proposed a [long time ago](https://github.com/rspec/rspec-core/issues/456) but didn't get around to making a PR.

This looks good to me (I agree with the comment Tute left above)

File: templates/Gemfile.erb [hunk 2]
```diff
@@ -39,7 +39,7 @@ group :development, :test do
   gem "dotenv-rails"
   gem "factory_girl_rails"
   gem "pry-rails"
-  gem "rspec-rails", "~> 3.1.0"
+  gem "rspec-rails", "~> 3.3.0"
```
Comment (JoelQ) [inline templates/Gemfile.erb]: Thoughts on using `"~> 3.3" so as to automatically pick up minor versions?

Comment (tute) [inline templates/Gemfile.erb]: Both ways result in the same behavior.

Comment (JoelQ) [inline templates/Gemfile.erb]: Won't `"~> 3.3.0"` update to `3.3.1` but not `3.4.0`

Comment (tute) [inline templates/Gemfile.erb]: I didn't get your "minor" mention. I was not aware that it works that way, I'm in favor of the change if that is the case.

Comment (derekprior) [inline templates/Gemfile.erb]: I'd be in favor of changing it to `>= 3.3.0` myself, but I think that's a change for a different PR.

**Comment by delphaber** [comment]
Hi, I think there aren't any tests for `spec_helper`and `rails_helper`. Maybe these lines could be useful:

```ruby
  it "adds rspec's helper files" do
    expect(File).to exist("#{project_path}/spec/spec_helper.rb")
    expect(File).to exist("#{project_path}/spec/rails_helper.rb")
  end
````

**Pull request closed without merging**